src/deploy/functions/params.ts

```diff
@@ -267,83 +396,271 @@ export async function resolveParams(
   if (!param.input) {
-    const defaultToText: TextInput<number> = { type: "text", text: {} };
+    const defaultToText: TextInput<number> = { type: "text" };
     param.input = defaultToText;
   }
+  let prompt: string;
 
   switch (param.input.type) {
     case "select":
-      throw new FirebaseError(
-        "Build specified int parameter " + param.name + " with unsupported input type 'select'"
-      );
+      prompt = `Select a value for ${param.label || param.name}:`;
+      if (param.description) {
+        prompt += ` \n(${param.description})`;
+      }
+      prompt += "\nSelect an option with the arrow keys, and use Enter to confirm your choice. ";
+      return promptSelect(prompt, param.input, resolvedDefault, (res: string) => +res);
     case "text":
     default:
-      let prompt = `Enter a value for ${param.label || param.name}:`;
+      prompt = `Enter an integer value for ${param.label || param.name}:`;
       if (param.description) {
         prompt += ` \n(${param.description})`;
       }
-      let res: number;
-      while (true) {
-        res = await promptOnce({
-          name: param.name,
-          type: "number",
-          default: resolvedDefault,
-          message: prompt,
-        });
-        if (Number.isInteger(res)) {
-          return res;
+      return promptText<number>(prompt, param.input, resolvedDefault, (res: string) => {
+        if (isNaN(+res)) {
+          return { message: `"${res}" could not be converted to a number.` };
+        }
+        if (res.includes(".")) {
+          return { message: `${res} is not an integer value.` };
         }
-        logger.error(`${param.label || param.name} must be an integer; retrying...`);
+        return +res;
+      });
+  }
+}
+
+async function promptResourceString(
+  prompt: string,
+  input: ResourceInput,
+  projectId: string,
+  resolvedDefault?: string
+): Promise<string> {
+  const notFound = new FirebaseError(`No instances of ${input.resource.type} found.`);
+  switch (input.resource.type) {
+    case "storage.googleapis.com/Bucket":
+      const buckets = await listBuckets(projectId);
+      if (buckets.length === 0) {
+        throw notFound;
       }
+      const forgedInput: SelectInput<string> = {
+        type: "select",
+        select: buckets.map((bucketName: string): SelectOptions<string> => {
+          return { label: bucketName, value: bucketName };
+        }),
+      };
+      return promptSelect<string>(prompt, forgedInput, resolvedDefault, (res: string) => res);
+    default:
+      logger.warn(
+        `Warning: unknown resource type ${input.resource.type}; defaulting to raw text input...`
+      );
+      return promptText<string>(prompt, { type: "text" }, resolvedDefault, (res: string) => res);
+  }
+}
+
+type retryInput = { message: string };
+async function promptText<T extends RawParamValue>(
+  prompt: string,
+  input: TextInput<T>,
+  resolvedDefault: T | undefined,
+  converter: (res: string) => T | retryInput
```

Comment:
I think it make sense to make this argument optional instead of having call pass identity function.

Reply:
Wouldn't that only make sense if T was `string` itself? There's no implicit way to make an arbitrary type out of a string, is there?

Reply:
oh good point. Didn't consider that aspect.